File: app/src/main/java/com/example/kampustangerang/Campus.java
package com.example.kampustangerang;

public class Campus {
    private String name;
    private String location;
    private String description;
    private String imageUrl;
    private String facilities;
    private String registrationDate;

    public Campus(String name, String location, String description, String imageUrl, String facilities, String registrationDate) {
        this.name = name;
        this.location = location;
        this.description = description;
        this.imageUrl = imageUrl;
        this.facilities = facilities;
        this.registrationDate = registrationDate;
    }

    public String getName() {
        return name;
    }

    public String getLocation() {
        return location;
    }

    public String getDescription() {
        return description;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getFacilities() {
        return facilities;
    }

    public String getRegistrationDate() {
        return registrationDate;
    }
}

File: app/src/main/java/com/example/kampustangerang/Campus.kt
package com.example.kampustangerang

import java.io.Serializable

data class Campus(
    val name: String,
    val location: String,
    val description: String,
    val imageUrl: String,
    val facilities: String,
    val registrationDate: String,
    val category: String = "Umum", // IPA, IPS, Bahasa, atau Umum
    val rector: String = "",
    val facultiesList: List<Faculty> = emptyList()
) : Serializable

data class Faculty(
    val name: String,
    val dean: String,
    val studyPrograms: List<StudyProgram>
) : Serializable

data class StudyProgram(
    val name: String,
    val headOfProgram: String
) : Serializable
